src/lib.rs

```diff
@@ -18,20 +19,70 @@ cfg_if! {
     }
 }
 
+struct Skin(DynamicImage);
+
+enum Layer {
+    Bottom,
+    Top,
+    Both,
+}
+
+enum BodyPart {
+    Head,
+    Body,
+    ArmLeft,
+    ArmRight,
+    LegLeft,
+    LegRight,
+}
+
+impl Skin {
+    fn get_part(&mut self, layer: &Layer, part: &BodyPart) -> DynamicImage {
+        match layer {
+            Layer::Both => {
+                let mut bottom = self.get_part(&Layer::Bottom, part);
+                let top = self.get_part(&Layer::Top, part);
+                imageops::overlay(&mut bottom, &top, 0, 0);
+                bottom
+            },
+            Layer::Bottom => {
+                match part {
+                    BodyPart::Head => self.0.crop(8, 8, 8, 8),
```

Comment:
Instead of using `DynamicImage::crop`, perhaps we could consider using `GenericImageView::view` instead, then overlay into a mutable image when we need it.

Reply:
This will make #2 much easier to implement as well.